test/e2e/user_workload_monitoring_test.go

```diff
@@ -935,31 +971,24 @@ func assertUWMFederateEndpoint(t *testing.T) {
 		}
 		defer cleanUp()
 
-		client := framework.NewPrometheusClient(
-			host,
-			token,
-			&framework.QueryParameterInjector{
-				Name:  "match[]",
-				Value: `up`,
-			},
-		)
-
-		resp, err := client.Do("GET", "/federate", nil)
+		err = federate(host)
 		if err != nil {
 			return err
 		}
-		defer resp.Body.Close()
 
-		b, err := ioutil.ReadAll(resp.Body)
+		r, err := factory.PrometheusUserWorkloadFederateRoute()
```

Comment:
I'd prefer if we didn't couple the e2e test with the manifests package since this has been the approach so far. In practice, the test should assume the name of the route.